pkg/controller/datavolume-controller.go

```diff
@@ -764,10 +764,10 @@ func (r *DatavolumeReconciler) emitFailureConditionEvent(dataVolume *cdiv1.DataV
 }
 
 func (r *DatavolumeReconciler) emitEvent(dataVolume *cdiv1.DataVolume, dataVolumeCopy *cdiv1.DataVolume, curPhase cdiv1.DataVolumePhase, originalCond []cdiv1.DataVolumeCondition, event *DataVolumeEvent) error {
-	// Only update the object if something actually changed in the status.
-	if !reflect.DeepEqual(dataVolume, dataVolumeCopy) {
-		if err := r.client.Update(context.TODO(), dataVolumeCopy); err != nil {
-			r.log.Error(err, "Unable to update datavolume", "name", dataVolumeCopy.Name)
+	// Update datavolume status if it changed
+	if !reflect.DeepEqual(dataVolume.Status, dataVolumeCopy.Status) {
```

Comment:
if we do not have subresource status enabled, does this really work as intended?  Spec will get updated when calling `Update`.

Also, this will not detect any changes to metadata.  I can think of at least one place where an annotation is added

Reply:
It won't work without subresource status enabled.
Do we want to update the spec in some flows?

Reply:
Also, I assume `Update` will update the spec so does this work as intended?  Is there an UpdateStatus call on controller runtime client?

Reply:
We want to update metadata which requires a call to `Update`

Reply:
https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/controller/datavolume-controller.go#L659-L664

Reply:
You could substitute any annotation with a status field, though.

I think this issue is perhaps best addressed in the webhook.  I assume that PVCs have the same issue.  I think they just ignore updates to the spec?

Reply:
That explains the test failures.
AFAIU, moving annotation to a status field is the right way to do it, but won't it introduce lots of noise?
Looking at the PVC code.